docs/examples/tensorflow/tensorflow-dataset-multigpu.ipynb

```diff
@@ -0,0 +1,311 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Using Tensorflow DALI plugin: DALI tf.data.Dataset with multiple GPUs\n",
+    "\n",
+    "### Overview\n",
+    "\n",
+    "This notebook is a comprehensive example on how to use DALI `tf.data.Dataset` with multiple GPUs. It is recommended to look into [single GPU example](tensorflow-dataset.ipynb) first to get up to speed with DALI dataset and how it can be used to train a neural network with custom model and training loop. This example is an extension of the single GPU version.\n",
+    "\n",
+    "Initially we define some parameters of the training and to create a DALI pipeline to read [MNIST]() dataset. This pipeline is able to shard dataset into multiple parts."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import nvidia.dali as dali\n",
+    "from nvidia.dali.pipeline import Pipeline\n",
+    "import nvidia.dali.ops as ops\n",
+    "import nvidia.dali.types as types\n",
+    "\n",
+    "import os\n",
+    "\n",
+    "import nvidia.dali.plugin.tf as dali_tf\n",
+    "import tensorflow.compat.v1 as tf\n",
+    "\n",
+    "tf.disable_eager_execution()\n",
+    "tf.reset_default_graph()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Path to MNIST dataset\n",
+    "data_path = os.path.join(os.environ['DALI_EXTRA_PATH'], 'db/MNIST/training/')\n",
+    "\n",
+    "BATCH_SIZE = 64\n",
+    "DROPOUT = 0.2\n",
+    "IMAGE_SIZE = 28\n",
+    "NUM_CLASSES = 10\n",
+    "HIDDEN_SIZE = 128\n",
+    "EPOCHS = 5\n",
+    "ITERATIONS = 100\n",
+    "num_gpus = 2"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class MnistPipeline(Pipeline):\n",
+    "    def __init__(self, batch_size, device_id=0, shard_id=0, num_shards=1, num_threads=4, seed=0):\n",
+    "        super(MnistPipeline, self).__init__(\n",
+    "            batch_size, num_threads, device_id, seed)\n",
+    "        self.reader = ops.Caffe2Reader(path=data_path, random_shuffle=True, shard_id=0, num_shards=num_shards)\n",
+    "        self.decode = ops.ImageDecoder(\n",
+    "            device='mixed',\n",
+    "            output_type=types.GRAY)\n",
+    "        self.cmn = ops.CropMirrorNormalize(\n",
+    "            device='gpu',\n",
+    "            output_dtype=types.FLOAT,\n",
+    "            image_type=types.GRAY,\n",
+    "            mean=[0.],\n",
```

Comment:
You can omit it and use the default value of mean and std.

Reply:
Done